contracts/handlers/ERC721Handler.sol

```diff
@@ -171,24 +171,24 @@ contract ERC721Handler is IDepositExecute, HandlerHelpers, ERC721Safe {
         metadata                        length      uint256    bytes    (96 + len(destinationRecipientAddress)) - (96 + len(destinationRecipientAddress) + 32)
         metadata                                      bytes    bytes    (96 + len(destinationRecipientAddress) + 32) - END
      */
-    function executeDeposit(bytes memory data) public override _onlyBridge {
+    function executeDeposit(bytes calldata data) external override _onlyBridge {
         uint256         tokenID;
         bytes32         resourceID;
         bytes  memory   destinationRecipientAddress;
         bytes  memory   metaData;
 
         assembly {
-            tokenID                        := mload(add(data, 0x20))
-            resourceID                     := mload(add(data, 0x40))
+            tokenID    := calldataload(0x44)
+            resourceID := calldataload(0x64)
 
 
             // set up destinationRecipientAddress
-            destinationRecipientAddress     := mload(0x40)              // load free memory pointer
-            let lenDestinationRecipientAddress  := mload(add(data, 0x60))
+            destinationRecipientAddress        := mload(0x40)              // load free memory pointer
+            let lenDestinationRecipientAddress := calldataload(0x84)
 
             // set up metaData
-            let lenMeta    := mload(add(data, add(0x80, lenDestinationRecipientAddress)))
-
+            // let lenMeta := mload(add(data, add(0x80, lenDestinationRecipientAddress)))
+            let lenMeta := calldataload(add(0x24, add(0x80, lenDestinationRecipientAddress)))
```

Comment:
I think we can add 0x24 and 0x80 ourselves and add a comment indicating how the number came to me ie position of data in calldata + position of lenMeta in data.